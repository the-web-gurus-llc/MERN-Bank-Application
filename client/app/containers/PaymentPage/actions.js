/*
 *
 * PaymentPage actions
 *
 */

import {
  GET_BILLS_REQUEST,
  GET_BILLS_SUCCESS,
  GET_BILLS_ERROR,
  SEARCH_RECIPIENT_REQUEST,
  SEARCH_RECIPIENT_SUCCESS,
  SEARCH_RECIPIENT_ERROR,
  CREATE_TRANSACTION_REQUEST,
  CREATE_TRANSACTION_SUCCESS,
  CREATE_TRANSACTION_ERROR,
  GET_AUTHORIZATION_KEY_REQUEST,
  GET_AUTHORIZATION_KEY_SUCCESS,
  GET_AUTHORIZATION_KEY_ERROR,
  CONFIRM_TRANSACTION_REQUEST,
  CONFIRM_TRANSACTION_SUCCESS,
  CONFIRM_TRANSACTION_INCORRECT,
  CHECK_RECIPIENT,
  CHECK_RECIPIENT_CORRECT,
  CHECK_RECIPIENT_INCORRECT,
  SELECT_SENDER_BILL,
} from './constants';

export function getBillsAction() {
  return {
    type: GET_BILLS_REQUEST,
  };
}

export function getBillsSuccessAction(bills) {
  return {
    type: GET_BILLS_SUCCESS,
    bills,
  };
}

export function getBillsErrorAction(error) {
  return {
    type: GET_BILLS_ERROR,
    error,
  };
}

export function searchRecipientAction(value) {
  return {
    type: SEARCH_RECIPIENT_REQUEST,
    value: value.replace(/ /g, ''),
  };
}

export function searchRecipientSuccessAction(recipients) {
  return {
    type: SEARCH_RECIPIENT_SUCCESS,
    recipients,
  };
}

export function searchRecipientErrorAction(error) {
  return {
    type: SEARCH_RECIPIENT_ERROR,
    error,
  };
}

export function checkRecipientAction() {
  return {
    type: CHECK_RECIPIENT,
  };
}

export function checkRecipientCorrectAction() {
  return {
    type: CHECK_RECIPIENT_CORRECT,
  };
}

export function checkRecipientIncorrectAction(error) {
  return {
    type: CHECK_RECIPIENT_INCORRECT,
    error,
  };
}

export function createTransactionAction() {
  return {
    type: CREATE_TRANSACTION_REQUEST,
  };
}

export function createTransactionSuccessAction(uuid) {
  return {
    type: CREATE_TRANSACTION_SUCCESS,
    uuid,
  };
}

export function createTransactionErrorAction(error) {
  return {
    type: CREATE_TRANSACTION_ERROR,
    error,
  };
}

export function getAuthorizationKeyAction() {
  return {
    type: GET_AUTHORIZATION_KEY_REQUEST,
  };
}

export function getAuthorizationKeySuccessAction(authorizationKey) {
  return {
    type: GET_AUTHORIZATION_KEY_SUCCESS,
    authorizationKey,
  };
}

export function getAuthorizationKeyErrorAction(error) {
  return {
    type: GET_AUTHORIZATION_KEY_ERROR,
    error,
  };
}

export function confirmTransactionAction(snippets) {
  return {
    type: CONFIRM_TRANSACTION_REQUEST,
    snippets,
  };
}

export function confirmTransactionSuccessAction(snippets) {
  return {
    type: CONFIRM_TRANSACTION_SUCCESS,
    snippets,
  };
}

export function confirmTransactionIncorrectAction(error) {
  return {
    type: CONFIRM_TRANSACTION_INCORRECT,
    error,
  };
}

export function selectSenderBillAction(uuid) {
  return {
    type: SELECT_SENDER_BILL,
    uuid,
  };
}
