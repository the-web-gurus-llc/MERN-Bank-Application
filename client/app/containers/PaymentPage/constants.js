/*
 *
 * PaymentPage constants
 *
 */

export const GET_BILLS_REQUEST = 'app/PaymentPage/GET_BILLS_REQUEST';
export const GET_BILLS_SUCCESS = 'app/PaymentPage/GET_BILLS_SUCCESS';
export const GET_BILLS_ERROR = 'app/PaymentPage/GET_BILLS_ERROR';

export const SEARCH_RECIPIENT_REQUEST =
  'app/PaymentPage/SEARCH_RECIPIENT_REQUEST';
export const SEARCH_RECIPIENT_SUCCESS =
  'app/PaymentPage/SEARCH_RECIPIENT_SUCCESS';
export const SEARCH_RECIPIENT_ERROR = 'app/PaymentPage/SEARCH_RECIPIENT_ERROR';

export const CHECK_RECIPIENT = 'app/PaymentPage/CHECK_RECIPIENT';
export const CHECK_RECIPIENT_CORRECT =
  'app/PaymentPage/CHECK_RECIPIENT_CORRECT';
export const CHECK_RECIPIENT_INCORRECT =
  'app/PaymentPage/CHECK_RECIPIENT_INCORRECT';

export const CREATE_TRANSACTION_REQUEST =
  'app/PaymentPage/CREATE_TRANSACTION_REQUEST';
export const CREATE_TRANSACTION_SUCCESS =
  'app/PaymentPage/CREATE_TRANSACTION_SUCCESS';
export const CREATE_TRANSACTION_ERROR =
  'app/PaymentPage/CREATE_TRANSACTION_ERROR';

export const GET_AUTHORIZATION_KEY_REQUEST =
  'app/PaymentPage/GET_AUTHORIZATION_KEY_REQUEST';
export const GET_AUTHORIZATION_KEY_SUCCESS =
  'app/PaymentPage/GET_AUTHORIZATION_KEY_SUCCESS';
export const GET_AUTHORIZATION_KEY_ERROR =
  'app/PaymentPage/GET_AUTHORIZATION_KEY_ERROR';

export const CONFIRM_TRANSACTION_REQUEST =
  'app/PaymentPage/CONFIRM_TRANSACTION_REQUEST';
export const CONFIRM_TRANSACTION_SUCCESS =
  'app/PaymentPage/CONFIRM_TRANSACTION_SUCCESS';
export const CONFIRM_TRANSACTION_INCORRECT =
  'app/PaymentPage/CONFIRM_TRANSACTION_INCORRECT';

export const SELECT_SENDER_BILL = 'app/PaymentPage/SELECT_SENDER_BILL';
