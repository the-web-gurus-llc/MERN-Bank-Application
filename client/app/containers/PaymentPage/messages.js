/*
 * PaymentPage Messages
 *
 * This contains all the text for the PaymentPage container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.PaymentPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the PaymentPage container!',
  },
  payment: {
    id: `${scope}.payment`,
    defaultMessage: 'Payment',
  },
  billNotValid: {
    id: `${scope}.bill.not.valid`,
    defaultMessage: 'The bill number entered is not valid.',
  },
  authorizationKeyIncorrect: {
    id: `${scope}.authorization.key.incorrect`,
    defaultMessage: 'Invalid code',
  },
});
