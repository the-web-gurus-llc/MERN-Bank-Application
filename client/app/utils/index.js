export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
