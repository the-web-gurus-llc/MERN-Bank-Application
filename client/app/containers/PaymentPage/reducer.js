/*
 *
 * PaymentPage reducer
 *
 */

import produce from 'immer';
import { LOCATION_CHANGE } from 'connected-react-router';
import {
  LOGOUT_SUCCESS,
  LOGOUT_ERROR,
  CHANGE_INPUT,
  CHANGE_INPUT_NUMBER,
  NEXT_STEP,
  PREVIOUS_STEP,
} from 'containers/App/constants';
import { formatBill } from 'helpers';
import { routes } from 'utils';
import {
  SELECT_SENDER_BILL,
  GET_BILLS_SUCCESS,
  SEARCH_RECIPIENT_SUCCESS,
  GET_AUTHORIZATION_KEY_SUCCESS,
  CHECK_RECIPIENT_CORRECT,
  CREATE_TRANSACTION_SUCCESS,
  CONFIRM_TRANSACTION_SUCCESS,
} from './constants';

export const initialState = {
  transaction: '',
  bills: [],
  recipients: [],
  senderBill: {},
  recipientBill: {},
  amountMoney: '',
  transferTitle: '',
  authorizationKey: '',
  currentStep: 0,
  hasCreatedTransaction: false,
  hasConfirmedTransaction: false,
};

/* eslint-disable default-case, no-param-reassign, consistent-return */
const paymentPageReducer = produce((draft, action) => {
  if (window.location.pathname === routes.payment.path) {
    switch (action.type) {
      case CHANGE_INPUT:
        if (action.name === 'recipientAccountBillNumber') {
          draft.recipientBill =
            draft.recipients?.find(
              (bill) =>
                bill.accountBillNumber.replace(/ /g, '') ===
                action.value.trim(),
            ) || '';
        } else {
          draft[action.name] = action.value.trim();
        }
        break;
      case CHANGE_INPUT_NUMBER:
        draft[action.name] = action.value || initialState.amountMoney;
        break;
      case CHECK_RECIPIENT_CORRECT:
      case NEXT_STEP:
        draft.currentStep += 1;
        break;
      case PREVIOUS_STEP:
        draft.currentStep -= 1;
        draft.hasCreatedTransaction = false;
        draft.authorizationKey = initialState.authorizationKey;
        break;
    }
  }

  switch (action.type) {
    case GET_AUTHORIZATION_KEY_SUCCESS:
      draft.authorizationKey = action.authorizationKey;
      break;
    case GET_BILLS_SUCCESS:
      draft.bills = action.bills.map(formatBill);
      break;
    case SEARCH_RECIPIENT_SUCCESS:
      draft.recipients = action.recipients.map(formatBill);
      break;
    case SELECT_SENDER_BILL:
      draft.senderBill =
        draft.bills?.find((bill) => bill.uuid === action.uuid) || '';
      break;
    case CREATE_TRANSACTION_SUCCESS:
      draft.hasCreatedTransaction = true;
      draft.transaction = action.uuid;
      break;
    case CONFIRM_TRANSACTION_SUCCESS:
      draft.hasConfirmedTransaction = true;
      break;
    case LOGOUT_SUCCESS:
    case LOGOUT_ERROR:
    case LOCATION_CHANGE:
      return initialState;
  }
}, initialState);

export default paymentPageReducer;
