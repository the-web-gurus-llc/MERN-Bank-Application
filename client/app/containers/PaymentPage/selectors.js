import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the paymentPage state domain
 */

const selectPaymentPageDomain = (state) => state.paymentPage || initialState;

/**
 * Other specific selectors
 */

/**
 * Default selector used by PaymentPage
 */

const makeSelectBills = () =>
  createSelector(selectPaymentPageDomain, (substate) => substate.bills);

const makeSelectRecipients = () =>
  createSelector(selectPaymentPageDomain, (substate) => substate.recipients);

const makeSelectSenderBill = () =>
  createSelector(selectPaymentPageDomain, (substate) => substate.senderBill);

const makeSelectSenderBillUuid = () =>
  createSelector(
    selectPaymentPageDomain,
    (substate) => substate.senderBill?.uuid,
  );

const makeSelectRecipientBill = () =>
  createSelector(selectPaymentPageDomain, (substate) => substate.recipientBill);

const makeSelectRecipientBillUuid = () =>
  createSelector(
    selectPaymentPageDomain,
    (substate) => substate.recipientBill?.uuid,
  );

const makeSelectAmountMoney = () =>
  createSelector(selectPaymentPageDomain, (substate) => substate.amountMoney);

const makeSelectTransferTitle = () =>
  createSelector(selectPaymentPageDomain, (substate) => substate.transferTitle);

const makeSelectAuthorizationKey = () =>
  createSelector(
    selectPaymentPageDomain,
    (substate) => substate.authorizationKey,
  );

const makeSelectCurrentStep = () =>
  createSelector(selectPaymentPageDomain, (substate) => substate.currentStep);

const makeSelectHasCreatedTransaction = () =>
  createSelector(
    selectPaymentPageDomain,
    (substate) => substate.hasCreatedTransaction,
  );

const makeSelectHasConfirmedTransaction = () =>
  createSelector(
    selectPaymentPageDomain,
    (substate) => substate.hasConfirmedTransaction,
  );

const makeSelectTransaction = () =>
  createSelector(selectPaymentPageDomain, (substate) => substate.transaction);

export {
  selectPaymentPageDomain,
  makeSelectBills,
  makeSelectRecipients,
  makeSelectSenderBill,
  makeSelectRecipientBill,
  makeSelectAmountMoney,
  makeSelectTransferTitle,
  makeSelectAuthorizationKey,
  makeSelectCurrentStep,
  makeSelectHasCreatedTransaction,
  makeSelectHasConfirmedTransaction,
  makeSelectTransaction,
  makeSelectSenderBillUuid,
  makeSelectRecipientBillUuid,
};
