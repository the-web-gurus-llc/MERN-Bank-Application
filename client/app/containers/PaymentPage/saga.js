import React from 'react';
import { takeLatest, delay, call, put, select } from 'redux-saga/effects';
import { api, request, routes } from 'utils';
import { push } from 'connected-react-router';
import { notification } from 'antd';
import { makeSelectLocale } from 'providers/LanguageProvider/selectors';
import {
  makeSelectToken,
  makeSelectIsCollapsedSidebar,
} from 'containers/App/selectors';
import {
  makeSelectRecipients,
  makeSelectRecipientBill,
  makeSelectRecipientBillUuid,
  makeSelectSenderBillUuid,
  makeSelectAmountMoney,
  makeSelectTransferTitle,
  makeSelectTransaction,
  makeSelectAuthorizationKey,
} from 'containers/PaymentPage/selectors';
import { FormattedMessage } from 'react-intl';
import {
  GET_BILLS_REQUEST,
  SEARCH_RECIPIENT_REQUEST,
  CHECK_RECIPIENT,
  CREATE_TRANSACTION_REQUEST,
  CONFIRM_TRANSACTION_REQUEST,
  GET_AUTHORIZATION_KEY_REQUEST,
} from './constants';
import {
  getBillsSuccessAction,
  getBillsErrorAction,
  searchRecipientSuccessAction,
  searchRecipientErrorAction,
  checkRecipientCorrectAction,
  checkRecipientIncorrectAction,
  createTransactionSuccessAction,
  getAuthorizationKeySuccessAction,
  // getAuthorizationKeyErrorAction,
  confirmTransactionIncorrectAction,
  confirmTransactionSuccessAction,
} from './actions';
import messages from './messages';
import { numberValidation } from '../../helpers';

export function* getBills() {
  const { accessToken } = yield select(makeSelectToken());
  const requestURL = api.bills();
  const requestParameters = {
    method: 'GET',
    headers: { Authorization: `Bearer ${accessToken}` },
  };

  try {
    const { data } = yield call(request, requestURL, requestParameters);
    yield put(getBillsSuccessAction(data));
  } catch (error) {
    yield put(getBillsErrorAction(error));

    switch (error.statusCode) {
      case 401:
        yield put(push(routes.login.path));
        break;
      default:
        yield put(push(routes.login.path));
        break;
    }
  }
}

export function* searchRecipient({ value }) {
  const { accessToken } = yield select(makeSelectToken());
  const requestURL = api.bills('search')(value);
  const requestParameters = {
    method: 'GET',
    headers: { Authorization: `Bearer ${accessToken}` },
  };

  if (!numberValidation(value)) {
    return;
  }

  if (value.length > 1) {
    yield delay(300);
  }

  try {
    const { data } = yield call(request, requestURL, requestParameters);
    yield put(searchRecipientSuccessAction(data));
  } catch (error) {
    yield put(searchRecipientErrorAction(error));

    switch (error.statusCode) {
      case 401:
        yield put(push(routes.login.path));
        break;
      default:
        yield put(push(routes.login.path));
        break;
    }
  }
}

export function* checkRecipient() {
  const recipients = yield select(makeSelectRecipients());
  const recipientBill = yield select(makeSelectRecipientBill());
  const isExist = recipients.find(({ uuid }) => uuid === recipientBill.uuid);

  if (isExist) {
    yield put(checkRecipientCorrectAction());
  } else {
    yield put(
      checkRecipientIncorrectAction(
        <FormattedMessage {...messages.billNotValid} />,
      ),
    );
  }
}

export function* createTransaction() {
  const { accessToken } = yield select(makeSelectToken());
  const locale = yield select(makeSelectLocale());
  const senderBill = yield select(makeSelectSenderBillUuid());
  const recipientBill = yield select(makeSelectRecipientBillUuid());
  const amountMoney = yield select(makeSelectAmountMoney());
  const transferTitle = yield select(makeSelectTransferTitle());
  const requestURL = api.transactions('create');
  const requestParameters = {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Bearer ${accessToken}`,
    },
    body: JSON.stringify({
      amountMoney,
      transferTitle,
      senderBill,
      recipientBill,
      locale,
    }),
  };

  try {
    const { uuid } = yield call(request, requestURL, requestParameters);
    yield put(createTransactionSuccessAction(uuid));
  } catch (error) {
    // yield put(createTransactionErrorAction(error));
    // switch (error.statusCode) {
    //   case 401:
    //     yield put(push(routes.login.path));
    //     break;
    //   default:
    //     yield put(push(routes.login.path));
    //     break;
    // }
  }
}

export function* getAuthorizationKey() {
  const { accessToken } = yield select(makeSelectToken());
  const transaction = yield select(makeSelectTransaction());
  const requestURL = api.transactions('authorizationKey')(transaction);
  const requestParameters = {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Bearer ${accessToken}`,
    },
  };

  try {
    const { authorizationKey } = yield call(
      request,
      requestURL,
      requestParameters,
    );
    yield put(getAuthorizationKeySuccessAction(authorizationKey));
  } catch (error) {
    // yield put(getAuthorizationKeyErrorAction(error));
    // switch (error.statusCode) {
    //   case 401:
    //     yield put(push(routes.login.path));
    //     break;
    //   default:
    //     yield put(push(routes.login.path));
    //     break;
    // }
  }
}

export function* confirmTransaction({ snippets }) {
  const { accessToken } = yield select(makeSelectToken());
  const authorizationKey = yield select(makeSelectAuthorizationKey());
  const isCollapsedSidebar = yield select(makeSelectIsCollapsedSidebar());
  const requestURL = api.transactions('confirm');
  const requestParameters = {
    method: 'PATCH',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Bearer ${accessToken}`,
    },
    body: JSON.stringify({ authorizationKey }),
  };
  const style = { width: 400, marginLeft: isCollapsedSidebar ? 80 : 250 };
  const placement = 'bottomLeft';

  try {
    yield call(request, requestURL, requestParameters);
    yield put(confirmTransactionSuccessAction());

    notification.success({
      message: snippets.success.title,
      description: snippets.success.description,
      style,
      placement,
    });

    yield put(push(routes.dashboard.path));
  } catch (error) {
    let message;

    switch (error.statusCode) {
      case 404:
        message = <FormattedMessage {...messages.authorizationKeyIncorrect} />;
        break;
      default:
        break;
    }

    yield put(confirmTransactionIncorrectAction(message));
  }
}

export default function* paymentPageSaga() {
  yield takeLatest(GET_BILLS_REQUEST, getBills);
  yield takeLatest(SEARCH_RECIPIENT_REQUEST, searchRecipient);
  yield takeLatest(CHECK_RECIPIENT, checkRecipient);
  yield takeLatest(CREATE_TRANSACTION_REQUEST, createTransaction);
  yield takeLatest(GET_AUTHORIZATION_KEY_REQUEST, getAuthorizationKey);
  yield takeLatest(CONFIRM_TRANSACTION_REQUEST, confirmTransaction);
}
