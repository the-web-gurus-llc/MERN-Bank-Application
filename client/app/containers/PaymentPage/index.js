/**
 *
 * PaymentPage
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet-async';
import Footer from 'components/Footer';
import { useInjectReducer, useInjectSaga } from 'redux-injectors';
import PaymentForm from 'components/App/PaymentForm';
import { FormattedMessage } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import { makeSelectUser } from 'containers/App/selectors';
import { useSelector } from 'react-redux';
import { getAlertCount } from 'helpers';
import reducer from './reducer';
import saga from './saga';
import messages from './messages';

const key = 'paymentPage';
const stateSelector = createStructuredSelector({
  user: makeSelectUser(),
});

export default function PaymentPage() {
  useInjectReducer({ key, reducer });
  useInjectSaga({ key, saga });

  const { user } = useSelector(stateSelector);

  return (
    <>
      <FormattedMessage {...messages.payment}>
        {(title) => <Helmet title={`${getAlertCount(user)} ${title}`} />}
      </FormattedMessage>

      <PaymentForm />

      <Footer />
    </>
  );
}
