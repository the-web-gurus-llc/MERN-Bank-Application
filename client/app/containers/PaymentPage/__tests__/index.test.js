import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { HelmetProvider } from 'react-helmet-async';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import configureStore from 'utils/configureStore';
import { DEFAULT_LOCALE } from 'utils/locales';
import PaymentPage from '../index';
import 'utils/__tests__/__mocks__/matchMedia';

describe('<PaymentPage />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render its PaymentPage', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ConnectedRouter history={history}>
            <HelmetProvider>
              <PaymentPage />
            </HelmetProvider>
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );

    expect(firstChild).toMatchSnapshot();
  });
});
