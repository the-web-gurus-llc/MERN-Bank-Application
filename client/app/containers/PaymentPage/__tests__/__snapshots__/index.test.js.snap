// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentPage /> should render its PaymentPage 1`] = `
<div
  class="ant-steps ant-steps-horizontal Stepsstyle__StyledSteps-cszd7q-0 gvgAqU ant-steps-label-horizontal"
>
  <div
    class="ant-steps-item ant-steps-item-process ant-steps-item-active"
  >
    <div
      class="ant-steps-item-container"
    >
      <div
        class="ant-steps-item-tail"
      />
      <div
        class="ant-steps-item-icon"
      >
        <span
          class="ant-steps-icon"
        >
          1
        </span>
      </div>
      <div
        class="ant-steps-item-content"
      >
        <div
          class="ant-steps-item-title"
        >
          <span>
            Bill
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-steps-item ant-steps-item-wait"
  >
    <div
      class="ant-steps-item-container"
    >
      <div
        class="ant-steps-item-tail"
      />
      <div
        class="ant-steps-item-icon"
      >
        <span
          class="ant-steps-icon"
        >
          2
        </span>
      </div>
      <div
        class="ant-steps-item-content"
      >
        <div
          class="ant-steps-item-title"
        >
          <span>
            Recipient
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-steps-item ant-steps-item-wait"
  >
    <div
      class="ant-steps-item-container"
    >
      <div
        class="ant-steps-item-tail"
      />
      <div
        class="ant-steps-item-icon"
      >
        <span
          class="ant-steps-icon"
        >
          3
        </span>
      </div>
      <div
        class="ant-steps-item-content"
      >
        <div
          class="ant-steps-item-title"
        >
          <span>
            Amount of money
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-steps-item ant-steps-item-wait"
  >
    <div
      class="ant-steps-item-container"
    >
      <div
        class="ant-steps-item-tail"
      />
      <div
        class="ant-steps-item-icon"
      >
        <span
          class="ant-steps-icon"
        >
          4
        </span>
      </div>
      <div
        class="ant-steps-item-content"
      >
        <div
          class="ant-steps-item-title"
        >
          <span>
            Transfer title
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-steps-item ant-steps-item-wait"
  >
    <div
      class="ant-steps-item-container"
    >
      <div
        class="ant-steps-item-tail"
      />
      <div
        class="ant-steps-item-icon"
      >
        <span
          class="ant-steps-icon"
        >
          5
        </span>
      </div>
      <div
        class="ant-steps-item-content"
      >
        <div
          class="ant-steps-item-title"
        >
          <span>
            Confirm data
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
